### DISCUSSION TITLE
Why does DeriveEntityModel needs the struct to be named Model?

### DISCUSSION THREAD

**Discussion by Drakota** [SeaQL/sea-orm#515] [General]
I feel like this is a weird design decision especially since renaming with the "as" keyword doesn't change it's name when print-debugging and it makes it very hard to tell what Model you have in a heterogeneous collection.

**Comment by tyt2y3** [comment]
You are right! may be we should derive our `Debug` impl instead of relying on the built-in.
Then we are able to correctly pretty print it.
The reason behind that was Entity works at a module level, and it provides standardized paths to access it's types using `Model` `Entity` `Column` etc
